src/runtime/HalideRuntime.h

```diff
@@ -1198,7 +1198,7 @@ typedef struct halide_buffer_t {
      * this both adds clarity to code and will facilitate moving to
      * another representation for bounds query arguments. */
     HALIDE_ALWAYS_INLINE bool is_bounds_query() const {
-        return host == NULL;
+        return host == NULL && device == 0;
```

Comment:
Should we add `assert(!(host == NULL && device != 0))` since that is UB?

Reply:
I'm not comfortable with putting such an assert here as it is a totally random place to do that check. Also this state is moving toward defined behavior as passing device only allocations is fairly useful. (Though in a unified address space world, one ends up with combined allocations, not host == nullptr ones.)